src/Components/Alert/Alert_Model.php

```diff
@@ -13,11 +13,13 @@ class Alert_Model {
 	protected ?WP_Post $active_alert;
 	protected Alert_Settings $settings;
 	protected Alert_Rule_Manager $rule_manager;
+	protected Color_Options_Manager $options_manager;
 
-	public function __construct( Alert_Settings $settings, Alert_Rule_Manager $rule_manager ) {
-		$this->settings     = $settings;
-		$this->active_alert = $this->assign_current_alert();
-		$this->rule_manager = $rule_manager;
+	public function __construct( Alert_Settings $settings, Alert_Rule_Manager $rule_manager, Alert_Color_Options $options_manager ) {
```

Comment:
We should be passing the interface and not the concrete class, otherwise another developer could never override what gets passed in here without manually modifying this class, which makes the interface useless.